Sources/Caching/Caching.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Caching.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/13/25.
+
+import Foundation
+
+/// An inteface representing a simple cache
+@_spi(Internal) public protocol Caching {
```

Comment:
Nit: We already have the `DeviceCache` class to store things in UserDefaults that doesn't conform to this protocol. Also, the naming convention in the repo for protocols is to usually name them in the format `name` + `Type` (like `DeviceCacheKeyType`, `StoreKit2PurchaseIntentListenerType`, `PurchasesType`, etc.). 

It's a bit clunky, but maybe we could name this protocol something like `DataCacheType`, `LargeDataCacheType`, or `FileCacheType`?